packages/@aws-cdk/aws-config/lib/rule.ts

```diff
@@ -0,0 +1,374 @@
+import events = require('@aws-cdk/aws-events');
+import iam = require('@aws-cdk/aws-iam');
+import lambda = require('@aws-cdk/aws-lambda');
+import { Construct, IResource, Resource, Token } from '@aws-cdk/cdk';
+import { CfnConfigRule } from './config.generated';
+
+/**
+ * A config rule.
+ */
+export interface IRule extends IResource {
+  /**
+   * The name of the rule.
+   * @attribute
+   */
+  readonly configRuleName: string;
+
+  /**
+   * The arn of the rule.
+   * @attribute
+   */
+  readonly configRuleArn?: string;
```

Comment:
Is this really optional?

Reply:
I don't need `configRuleArn`, `configRuleId` or `configRuleComplianceType` when working with an *imported* rule. And I don't directly see why a user importing a rule would have to specify those.

Using only `configRuleName`, I can set up the `onEvent()`, `onComplianceChange()` and `onReEvaluationStatus()` for both new and imported rules.

If remove those properties (`configRuleArn`, `configRuleId`, `configRuleComplianceType`) from the `IRule` interface, I'm faced with the new `awslint:attribute-tag` rule errors:
```
error: [awslint:attribute-tag:@aws-cdk/aws-config.CloudFormationStackDriftDetectionCheck.configRuleArn] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleArn
error: [awslint:attribute-tag:@aws-cdk/aws-config.ManagedRule.configRuleId] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleId
error: [awslint:attribute-tag:@aws-cdk/aws-config.ManagedRule.configRuleComplianceType] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleComplianceType
error: [awslint:attribute-tag:@aws-cdk/aws-config.ManagedRule.configRuleArn] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleArn
error: [awslint:attribute-tag:@aws-cdk/aws-config.CustomRule.configRuleId] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.CustomRule.configRuleId
error: [awslint:attribute-tag:@aws-cdk/aws-config.CustomRule.configRuleComplianceType] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.CustomRule.configRuleComplianceType
error: [awslint:attribute-tag:@aws-cdk/aws-config.CustomRule.configRuleArn] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.CustomRule.configRuleArn
error: [awslint:attribute-tag:@aws-cdk/aws-config.CloudFormationStackNotificationCheck.configRuleId] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleId
error: [awslint:attribute-tag:@aws-cdk/aws-config.CloudFormationStackNotificationCheck.configRuleComplianceType] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleComplianceType
error: [awslint:attribute-tag:@aws-cdk/aws-config.AccessKeysRotated.configRuleArn] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleArnerror: [awslint:attribute-tag:@aws-cdk/aws-config.AccessKeysRotated.configRuleComplianceType] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleComplianceType
error: [awslint:attribute-tag:@aws-cdk/aws-config.AccessKeysRotated.configRuleId] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleId
error: [awslint:attribute-tag:@aws-cdk/aws-config.CloudFormationStackNotificationCheck.configRuleArn] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleArn
error: [awslint:attribute-tag:@aws-cdk/aws-config.CloudFormationStackDriftDetectionCheck.configRuleComplianceType] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleComplianceType
error: [awslint:attribute-tag:@aws-cdk/aws-config.CloudFormationStackDriftDetectionCheck.configRuleId] attribute properties must have an "@attribute" doctag on:  @aws-cdk/aws-config.ManagedRule.configRuleId
```

If I add `/** @attribute */` on the abstract properties of the abstract class `RuleNew`, the linter is not happy. The only valid alternative is to add `/** @attribute */` both in `ManagedRule` and `CustomRule` properties.

What is the guideline here?

Reply:
Good point, that has to do with the base class being invisible. I have created an issue for it: https://github.com/awslabs/jsii/issues/510

For now, best I can suggest is to put the tag on the individual rules. It feels a little silly, but that seems better to me than making it optional in the interface for the purposes of it being absent for an imported resources -- the interface is exactly meant to define the interface that can be relied upon to always be there.

Reply:
I just tested, and it seems like it should be inherited properly from the  `RuleNew` class as long as you avoid adding docstrings to the subtype.

So if there are no docstrings on the inherited members, you can do the full docs on the base member.

Reply:
You mean like this? Can't get this to work...
```ts
/**
 * A new managed or custom rule.
 */
abstract class RuleNew extends RuleBase {
  /**
   * The arn of the rule.
   *
   * @attribute
   */
  public abstract readonly configRuleArn: string;
```

```ts
/**
 * A new managed rule.
 *
 * @resource AWS::Config::ConfigRule
 */
export class ManagedRule extends RuleNew {
  public readonly configRuleName: string;
```